modules/openapi-generator/src/main/java/org/openapitools/codegen/CodegenProperty.java

```diff
@@ -149,6 +150,7 @@
     public boolean isArray;
     public boolean isMap;
     public boolean isEnum;
+    public boolean isReferenceToEnum;
```

Comment:
In the sample provided in previous comment, the enumProperty1 would not have the isEnum as true, but it would have isReferenceToEnum as true. I also intend to not mix the usage when you define a property as an inline enum in OpenAPI